modules/dependencies.nix

```diff
@@ -0,0 +1,52 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.dependencies;
+
+  packages = {
+    ctags = "ctags";
+    curl = "curl";
+    direnv = "direnv";
+    distant = "distant";
+    fish = "fish";
+    gcc = "gcc";
+    gh = "gh";
+    git = "git";
+    go = "go";
+    lean = "lean4";
+    ledger = "ledger";
+    manix = "manix";
+    nodejs = "nodejs";
+    ripgrep = "ripgrep";
+    sd = "sd";
+    sed = "gnused";
+    texpresso = "texpresso";
+    tinymist = "tinymist";
+    treesitter = "tree-sitter";
+    typst = "typst";
+    ueberzug = "ueberzugpp";
+    websocat = "websocat";
+    which = "which";
+    yazi = "yazi";
+  };
+
+  mkDependencyOption = name: default: {
+    enable = lib.mkEnableOption "Add ${name} to dependencies.";
+
+    package = lib.mkPackageOption pkgs name { inherit default; };
+  };
+in
+{
+  options.dependencies = lib.mapAttrs mkDependencyOption packages;
+
+  config = {
+    extraPackages = lib.pipe cfg [
+      (lib.filterAttrs (_: p: p.enable))
+      (lib.mapAttrsToList (_: p: p.package))
+    ];
```

Comment:
Alt approach, since we never need the attr name here, we can be _slightly_ more simple & efficient by working with a list:
```nix
extraPackages = lib.pipe cfg [
  builtins.attrValues
  (builtins.filter (p: p.enable))
  (builtins.map (p: p.package))
];
```